crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1270,16 +1270,53 @@ fn handle_import_from_comment<'a>(
 fn handle_with_comment<'a>(
     comment: DecoratedComment<'a>,
     with_statement: &'a ast::StmtWith,
+    locator: &Locator,
 ) -> CommentPlacement<'a> {
-    if comment.line_position().is_end_of_line()
-        && with_statement.items.first().is_some_and(|with_item| {
-            with_statement.start() < comment.start() && comment.start() < with_item.start()
-        })
+    if comment.line_position().is_own_line() {
+        return CommentPlacement::Default(comment);
+    }
+
+    if with_statement.items.first().is_some_and(|with_item| {
+        with_statement.start() < comment.start() && comment.start() < with_item.start()
+    }) {
+        return CommentPlacement::dangling(comment.enclosing_node(), comment);
+    }
+
+    // If a comment trails a `WithItem` with an `as`, consider attaching it to the optional
+    // variable, rather than to the `WithItem` itself.
+    //
+    // For example, given:
+    // ```python
+    // with (
+    //     a
+    //     as
+    //     b  # comment
+    //     , c as d
+    // ): ...
+    // ```
+    //
+    // The `# comment` should be attached to the optional variable `b`, rather than to the
+    // `WithItem` itself.
+    if let Some(AnyNodeRef::WithItem(ast::WithItem {
+        context_expr: _,
+        optional_vars: Some(optional_vars),
+        range: _,
+    })) = comment.preceding_node()
     {
-        CommentPlacement::dangling(comment.enclosing_node(), comment)
-    } else {
-        CommentPlacement::Default(comment)
+        let tokenizer = SimpleTokenizer::new(
+            locator.contents(),
+            TextRange::new(optional_vars.end(), comment.start()),
+        );
+        if !tokenizer.skip_trivia().any(|token| {
+            token.kind() == SimpleTokenKind::RParen
+                || token.kind() == SimpleTokenKind::Comma
+                || token.kind() == SimpleTokenKind::Colon
+        }) {
```

Comment:
Does this work as intended if you have

```python
with (
	a as 
	(((b))	 # comment
    )
	, c as d
): ...

```

Reply:
So, that _already_ gets associated as a trailing comment on `b`. I believe because the range of the `WithItem` starts at the `a` and goes until the parenthesis _after_ the `b`. The case we care about is that in which the comment trails the range of the `WithItem`, like:


```python
with (
    a
    as
    # comment
    b  # leading comment
): ...
```

Which makes me think: perhaps this condition should ignore `RParen`?